File: GREEDY TECHNIQUE/QUESTION 5/sol.java
import java.util.*;

public class sol {
    public static void main(String args[]){
        Scanner scr = new Scanner(System.in);
        int n = scr.nextInt();
        ArrayList<Integer> ar = new ArrayList<Integer>(n);
        for(int i=0;i<n;i++)
        {
            ar.add(scr.nextInt());
        }
        int neg = -1;
        Collections.sort(ar, Collections.reverseOrder());
        for(int i=0;i<n;i++)
        {
            if(ar.get(i)<0)
            {
                neg = i;
                break;
            }
        }
        int prod1 = ar.get(0)*ar.get(1)*ar.get(2)*ar.get(3)*ar.get(4);
        int prod2 = ar.get(0)*ar.get(1)*ar.get(2)*ar.get(n-1)*ar.get(n-2);
        int prod3 = ar.get(0)*ar.get(n-1)*ar.get(n-2)*ar.get(n-3)*ar.get(n-4);
        int ans = (neg==-1)?prod1:(neg>=n-3)?Math.max(prod1,prod2):(neg>=n-4)?Math.max(prod1,Math.max(prod1,prod2)):0;
        System.out.println(ans);
    }
}

File: GREEDY TECHNIQUE/QUESTION 5/sol.py
n = int(input())
ar = list(map(int,input().split()))
ar.sort(reverse=True)
neg = -1
for i in range(n):
    if ar[i]<0:
        neg = i
        break
prod1 = ar[0]*ar[1]*ar[2]*ar[3]*ar[4]
prod2 = ar[0]*ar[1]*ar[2]*ar[n-1]*ar[n-2]
prod3 = ar[0]*ar[n-1]*ar[n-2]*ar[n-3]*ar[n-4]
ans = -1
if neg == -1:
    ans = prod1
elif neg >= 3:
    ans = max(prod1,prod2)
else:
    ans = max(prod1,max(prod2,prod3))
print(ans)